modelgauge/prompt_pipeline.py

```diff
@@ -1,14 +1,25 @@
 import csv
 from abc import abstractmethod, ABCMeta
 from collections import defaultdict
+from dataclasses import dataclass
 from typing import Iterable
 
 import diskcache  # type: ignore
 
 from modelgauge.pipeline import Source, Pipe, Sink, CachingPipe
 from modelgauge.prompt import TextPrompt
 from modelgauge.single_turn_prompt_response import PromptWithContext
-from modelgauge.sut import PromptResponseSUT, SUT
+from modelgauge.sut import PromptResponseSUT, SUT, SUTCompletion
+
+
+@dataclass
+class SutInteraction:
```

Comment:
Nice. Typing helps clarify things, and I like the name especially.

Reply:
Yay!